File: 05-while-loop/exercises-submitted-evaluated/03.vacation.js
function vacation(input) {
    index = 0;
    let tripCost = Number(input[index]);
    index++;
    let moneySaved = Number(input[index]);
    index++;
    let spendCounter = 0;
    let daysCounter = 0;
    let transactionType = input[index];
    index++;
    daysCounter++;
    let transactionAmount = Number(input[index]);
    index++;

    while (index <= input.length) {
        switch (transactionType) {
            case "spend": moneySaved -= transactionAmount; spendCounter++; break;
            case "save": moneySaved += transactionAmount; spendCounter = 0; break;
            default: break;
        }
        if (moneySaved < 0) {
            moneySaved = 0;
        }
        if (spendCounter === 5) {
            console.log(`You can't save the money.`);
            console.log(`${daysCounter}`);
            break;
        }
        if (moneySaved >= tripCost) {
            console.log(`You saved the money for ${daysCounter} days.`);
            break;
        }
        transactionType = input[index];
        index++;
        daysCounter++;
        transactionAmount = Number(input[index]);
        index++;
    }
}

vacation(["2000",
    "1000",
    "spend",
    "1200",
    "save",
    "2000"]);
vacation(["110",
    "60",
    "spend",
    "10",
    "spend",
    "10",
    "spend",
    "10",
    "spend",
    "10",
    "spend",
    "10"]);
vacation(["250",
    "150",
    "spend",
    "50",
    "spend",
    "50",
    "save",
    "100",
    "save",
    "100"]);

File: softuni-python-basics/05_while_loop/03_vacation.py
trip_cost = float(input())
initial_budget = float(input())
current_budget = initial_budget
consequent_spending_days = 0
total_days = 0

while current_budget < trip_cost:
    transaction_type = input()
    transaction_amount = float(input())
    total_days += 1
    if transaction_type == "spend":
        current_budget -= transaction_amount
        consequent_spending_days += 1
    elif transaction_type == "save":
        current_budget += transaction_amount
        consequent_spending_days = 0
    if current_budget <= 0:
        current_budget = 0
    if consequent_spending_days >= 5:
        print("You can't save the money.")
        print(f"{total_days}")
        break
if current_budget >= trip_cost:
    print(f"You saved the money for {total_days} days.")
